akka-stream/src/main/scala/akka/stream/impl/io/TlsModule.scala

```diff
@@ -29,7 +30,7 @@ private[stream] final case class TlsModule(plainIn: Inlet[SslTlsOutbound], plain
 
   override def toString: String = f"TlsModule($closing) [${System.identityHashCode(this)}%08x]"
 
-  override private[stream] def traversalBuilder = TraversalBuilder.atomic(this)
+  override private[stream] def traversalBuilder = TraversalBuilder.atomic(this).makeIsland(TlsModuleIslandTag)
```

Comment:
I would expect that this ensures that above we only ever get a `TlsModule` as the module in `materializeAtomic`.